### ISSUE TITLE
AutoMagicIO not working (permission issues?)

### ISSUE THREAD

**Issue by mih** [datalad/datalad#1112]
This command would fail due to missing annex'ed files:
````
$ python code/mk_ds.py 16 stupid                        
Traceback (most recent call last):
...
  File "/usr/lib/python2.7/dist-packages/nibabel/loadsave.py", line 40, in load
    raise FileNotFoundError("No such file: '%s'" % filename)
nibabel.py3k.FileNotFoundError: No such file: 'src/templatesntransforms/sub-16/spaghetti1/mask.nii.gz'
````

AutoMagicIO should come to the rescue...but doesn't:

````
$ python -m datalad code/mk_ds.py 16 stupid           
[INFO   ] Activating DataLad's AutoMagicIO 
[INFO   ] Running code of code/mk_ds.py 
[ERROR  ] <class 'git.exc.InvalidGitRepositoryError'>: /etc 
[WARNING] Failed proxying open with ('/home/mih/datalad/datalad.egg-info/PKG-INFO', 'rb'), {}: [Errno 13] Permission denied: '/home/mih/datalad/.git/config.lock' [util.py:_obtain_lock_or_raise:758] 
[WARNING] Failed proxying open with ('/home/mih/datalad/datalad.egg-info/PKG-INFO', 'rb'), {}: [Errno 13] Permission denied: '/home/mih/datalad/.git/config.lock' [util.py:_obtain_lock_or_raise:758] 
fatal: Not a valid object name src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz
fatal: Not a valid object name src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz
fatal: Not a valid object name src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz
fatal: Not a valid object name src/templatesntransforms/sub-16/spaghetti1/mask_insula.nii.gz
fatal: Not a valid object name src/templatesntransforms/sub-16/spaghetti1/mask_insula.nii.gz
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/mih/datalad/datalad/__main__.py", line 102, in <module>
    main()
  File "/home/mih/datalad/datalad/__main__.py", line 93, in main
    runctx(code, globs, globs)
  File "/home/mih/datalad/datalad/__main__.py", line 41, in runctx
    exec(cmd, globals, locals)
  File "code/mk_ds.py", line 42, in <module>
    runs = [load_run(subid, runid) for runid in ('1', '2', '3', '4')]
  File "code/mk_ds.py", line 19, in load_run
    ds = fmri_dataset(bold_fname, mask=mask_fname)
  File "/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py", line 258, in fmri_dataset
    maskimg = _load_anyimg(mask)
  File "/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py", line 396, in _load_anyimg
    data = _img2data(src)
  File "/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py", line 72, in _img2data
    img = nibabel.load(src)
  File "/usr/lib/python2.7/dist-packages/nibabel/loadsave.py", line 48, in load
    filename)
nibabel.filebasedimages.ImageFileError: Cannot work out file type of "src/templatesntransforms/sub-16/spaghetti1/mask_insula.nii.gz"
````

This is using a datalad that was installed by different user as the one running the command. But write permissions shouldn't be necessary, right?

FTR: A manual

 ````
$ datalad get src/templatesntransforms/sub-16/spaghetti1/mask_insula.nii.gz
````

works as expected.

**Comment by yarikoptic** [comment]
A bit confused on the source of the issue here, ie how the fact that datalad installed (how? Pip or apt-get or?) by a different user plays here

I think those    fatal: Not a valid object name.  Started to appear only recently and not sure yet why in this case - needed to be traced down.

**Comment by yarikoptic** [comment]
Note that to deal with it I would need reproducible example or better clue in what is going on, ATM @mih is the only one having this unique setup ;-) may be running with higher log level could at least point to the problem more discretely?

**Comment by mih** [comment]
More logs are below. The call obtained none of the 5 required files. A manual `get` does it every time.

Also, Why the connection to github (see the end)?

````
$ DATALAD_LOG_LEVEL=debug python -m datalad code/mk_ds.py 16 stupid                                                                       
[INFO   ] Activating DataLad's AutoMagicIO 
[INFO   ] Running code of code/mk_ds.py 
[DEBUG  ] Running: ['git', 'annex', 'version', '--raw'] 
[ERROR  ] <class 'git.exc.InvalidGitRepositoryError'>: /etc 
[DEBUG  ] Using existing Git repository at /home/mih/datalad 
[DEBUG  ] Running: ['git', 'config', '-z', '-l'] 
[WARNING] Failed proxying open with ('/home/mih/datalad/datalad.egg-info/PKG-INFO', 'rb'), {}: [Errno 13] Permission denied: '/home/mih/datalad/.git/config.lock' [util.py:_obtain_lock_or_raise:758] 
[DEBUG  ] Using existing Git repository at /home/mih/datalad 
[WARNING] Failed proxying open with ('/home/mih/datalad/datalad.egg-info/PKG-INFO', 'rb'), {}: [Errno 13] Permission denied: '/home/mih/datalad/.git/config.lock' [util.py:_obtain_lock_or_raise:758] 
[DEBUG  ] Using existing Git repository at /home/XXX/testballoon/src/aligned 
[DEBUG  ] Running: ['git', 'config', '-z', '-l'] 
[DEBUG  ] Initiating a new process for BatchedAnnex(annex_cmd=['info'], annex_options=['--bytes', '--json'], git_options=<<['-c', 'receive.autogc...>>, output_proc=<function>, path=<<'/home/XXX/...>>) 
fatal: Not a valid object name src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz
[DEBUG  ] Closing stdin of <subprocess.Popen object at 0x7f5cc9f94510> and waiting process to finish 
[DEBUG  ] Process <subprocess.Popen object at 0x7f5cc9f94510> has finished 
[DEBUG  ] Using existing Git repository at /home/XXX/testballoon/src/aligned 
[DEBUG  ] Running: ['git', 'config', '-z', '-l'] 
[DEBUG  ] Initiating a new process for BatchedAnnex(annex_cmd=['info'], annex_options=['--bytes', '--json'], git_options=<<['-c', 'receive.autogc...>>, output_proc=<function>, path=<<'/home/XXX/...>>) 
fatal: Not a valid object name src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz
[DEBUG  ] Closing stdin of <subprocess.Popen object at 0x7f5cc9f94690> and waiting process to finish 
[DEBUG  ] Process <subprocess.Popen object at 0x7f5cc9f94690> has finished 
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/mih/datalad/datalad/__main__.py", line 102, in <module>
    main()
  File "/home/mih/datalad/datalad/__main__.py", line 93, in main
    runctx(code, globs, globs)
  File "/home/mih/datalad/datalad/__main__.py", line 41, in runctx
    exec(cmd, globals, locals)
  File "code/mk_ds.py", line 42, in <module>
    runs = [load_run(subid, runid) for runid in ('1', '2', '3', '4')]
  File "code/mk_ds.py", line 19, in load_run
    ds = fmri_dataset(bold_fname, mask=mask_fname)
  File "/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py", line 254, in fmri_dataset
    imgdata, imghdr, img = _load_anyimg(samples, ensure=True, enforce_dim=4)
  File "/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py", line 396, in _load_anyimg
    data = _img2data(src)
  File "/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py", line 72, in _img2data
    img = nibabel.load(src)
  File "/usr/lib/python2.7/dist-packages/nibabel/loadsave.py", line 48, in load
    filename)
nibabel.filebasedimages.ImageFileError: Cannot work out file type of "src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz"
[DEBUG  ] Closing 1 SSH connections... 
[DEBUG  ] Running: ['ssh', '-O', 'stop', '-o', u'ControlPath=/home/XXX/.cache/datalad/sockets/github.com', u'github.com'] 
[DEBUG  ] Failed to run ['ssh', '-O', 'stop', '-o', u'ControlPath=/home/XXX/.cache/datalad/sockets/github.com', u'github.com'] under None. Exit code=255. out= err=Control socket connect(/home/XXX/.cache/datalad/sockets/github.com): No such file or directory

````

**Comment by yarikoptic** [comment]
fixed by #1116
